pkgs/by-name/fo/foxglove-studio/package.nix

```diff
@@ -0,0 +1,106 @@
+{
+  alsa-lib,
+  at-spi2-atk,
+  cairo,
+  cups,
+  dbus,
+  dpkg,
+  eudev,
+  expat,
+  fetchurl,
+  glib,
+  glibc,
+  gtk3,
+  lib,
+  libdrm,
+  libgcc,
+  libxkbcommon,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  stdenv,
+  systemd,
+  via,
+  xorg,
+}:
+stdenv.mkDerivation rec {
+  name = "foxglove-studio";
+  version = "2.18.1";
+
+  src = fetchurl {
+    url = "https://get.foxglove.dev/desktop/latest/foxglove-studio-latest-linux-amd64.deb?_gl=1*14ju825*_gcl_au*NzYzMDc1MzM1LjE3MzM4NTg1ODk.";
+    sha256 = "sha256-vHloPOmrr8HR9Su5s84xohh8uCpbG4D9eEqbp+jPsDc=";
+  };
+  sourceRoot = ".";
+  unpackPhase = ''
+    dpkg-deb -x $src .
+  '';
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  nativeBuildInputs = [
+    dpkg
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -R usr/share opt $out/
+    # fix the path in the desktop file
+    substituteInPlace \
+      $out/share/applications/foxglove-studio.desktop \
+      --replace /opt/ $out/opt/
+    # symlink the binary to bin/
+    ln -s "$out/opt/Foxglove_Studio/foxglove-studio" $out/bin/foxglove-studio
+
+    runHook postInstall
+  '';
+
+  preFixup = let
+    libPath = lib.makeLibraryPath [
+      alsa-lib
+      at-spi2-atk
+      cairo
+      cups
+      dbus
+      eudev
+      expat
+      glib
+      glibc
+      gtk3
+      libdrm
+      libgcc.libgcc
+      libxkbcommon
+      mesa
+      nspr
+      nss
+      pango
+      systemd
+      via
+      xorg.libX11
+      xorg.libXcomposite
+      xorg.libXdamage
+      xorg.libXext
+      xorg.libXfixes
+      xorg.libXrandr
+      xorg.libxcb
+    ];
+  in ''
+    patchelf \
+      --set-interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      --set-rpath "${libPath}:$out/opt/Foxglove Studio/" \
+      "$out/opt/Foxglove Studio/foxglove-studio"
```

Comment:
I tried but the binary has some own libs included in `opt/Foxglove Studio/`, hence the need to add `:$out/opt/Foxglove Studio/` to the `rpath`.
I could probably tell `autoPatchelfHook` to also look there right? 
But if it isn't needed I would prefer leaving it like this, since I've already done some testing.

Reply:
Hmm. Maybe that's fine then (I'm sure that someone more familiar with patchelf-ing would suggest something better)

Reply:
Most definitely! Its my first time haha